### PR TITLE
feat: decode Solidity revert reasons (CPL-175)

### PR DESCRIPTION
**Pull request by GTC6244**
## Summary

Adds a `decode_revert` utility that decodes Solidity revert reasons from raw EVM error data, replacing opaque "gas price issue" error messages with the actual contract error.

**Core decoder (lit-core):** Generic Solidity revert decoder handling `Error(string)` (0x08c379a0) and `Panic(uint256)` (0x4e487b71) with no ethers dependency. 10 unit tests covering happy paths, edge cases, and error boundaries.

**Contract wrapper (lit-api-server):** Decodes `AccountConfigErrors` custom errors (AccountDoesNotExist, InsufficientBalance, NoAccountAccess, etc.) using the ethers-generated ABI bindings, with fallback to the generic decoder.

**Wiring:** `send_transaction` and `add_group` simulation call now surface decoded revert reasons instead of raw contract errors.

## Test Coverage

```
Tests: 10 passed, 0 failed
Coverage: 10/13 code paths tested (77%)
  lit-core decoder: 9/9 paths tested
  lit-api-server wrapper: requires ethers mocking (integration-level)
```

## Pre-Landing Review

No blocking issues. Informational findings acknowledged:
- Read-only `.call().await?` sites not yet wired (acceptable for v1, can extend later)
- Error chain flattened to string (design choice for human-readable messages)

## TODOS

Related: "Server should return 403 for management permission denials" — this PR enables that work by making contract error types decodable, but does not implement the HTTP status mapping.

## Test plan
- [x] All lit-core decode_revert tests pass (10 tests)
- [x] lit-api-server compiles cleanly

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: lit-core/lit-core/src/utils/decode_revert.rs [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+
+    // Read string length from bytes 32..64 (big-endian u256, but only the low bytes matter).
+    let len_bytes = &payload[32..64];
+    let len = u64::from_be_bytes(len_bytes[24..32].try_into().ok()?) as usize;
+
+    if payload.len() < 64 + len {
+        return None;
+    }
+
+    let s = std::str::from_utf8(&payload[64..64 + len]).ok()?;
```
[the remark this answers is not in the document]
Comment (GTC6244) [reply lit-core/lit-core/src/utils/decode_revert.rs]: Fixed in 20abdfc. Now uses `checked_add` to safely compute `64 + len`, returning `None` on overflow instead of wrapping. Also added `usize::try_from` for the `u64 -> usize` conversion.

File: lit-core/lit-core/src/utils/decode_revert.rs [hunk 2]
```diff
@@ -0,0 +1,190 @@
...
+
+/// Decode the `Error(string)` ABI encoding:
+///   - bytes 0..32:  offset to string data (always 0x20)
+///   - bytes 32..64: string length
+///   - bytes 64..:   string content (padded to 32-byte boundary)
+fn decode_error_string(payload: &[u8]) -> Option<String> {
+    if payload.len() < 64 {
+        return None;
+    }
+
```
[the remark this answers is not in the document]
Comment (GTC6244) [reply lit-core/lit-core/src/utils/decode_revert.rs:43]: Fixed in 20abdfc. Now validates that the offset word's high 31 bytes are zero and the low byte is `0x20`. Returns `None` for non-standard offsets.

File: lit-core/lit-core/src/utils/decode_revert.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (GTC6244) [reply lit-core/lit-core/src/utils/decode_revert.rs]: Fixed in 20abdfc. Now rejects non-zero high 24 bytes in the length word, uses `usize::try_from(u64)` for safe conversion, and uses `checked_add` for the `64 + len` computation.

File: lit-core/lit-core/src/utils/decode_revert.rs [hunk 3]
```diff
@@ -0,0 +1,190 @@
...
+        0x21 => "conversion to invalid enum value",
+        0x22 => "access to incorrectly encoded storage byte array",
+        0x31 => "pop on empty array",
+        0x32 => "array index out of bounds",
+        0x41 => "allocation of too much memory",
+        0x51 => "called an uninitialized function pointer",
+        _ => return Some(format!("Panic(0x{code:02x})")),
+    };
+
+    Some(format!("Panic(0x{code:02x}): {description}"))
```
[the remark this answers is not in the document]
Comment (GTC6244) [reply lit-core/lit-core/src/utils/decode_revert.rs]: Fixed in 20abdfc. Now rejects non-zero high 24 bytes in the panic code word, and updated the comment to accurately describe the behavior.

File: lit-api-server/src/accounts/decode_revert.rs [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+    if let Some(data) = err.as_revert() {
+        // First, try contract-specific custom error decoding.
+        if let Ok(decoded) = AccountConfigErrors::decode(data) {
+            return format!("Contract error: {decoded}");
+        }
+
+        // Fall back to standard Error(string) / Panic(uint256).
+        if let Some(reason) = lit_core::utils::decode_revert::decode_revert(data) {
+            return format!("Revert: {reason}");
+        }
```
[the remark this answers is not in the document]
Comment (GTC6244) [reply lit-api-server/src/accounts/decode_revert.rs]: Fixed in 20abdfc. Reordered `decode_contract_revert` to try the lit-core generic decoder (Error/Panic) first, before `AccountConfigErrors::decode`. Standard revert strings now consistently get the `Revert: ...` label.

File: lit-api-server/src/accounts/decode_revert.rs [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+
+        // Unknown custom error — show the hex.
+        return format!("Unknown revert data: 0x{}", hex::encode(data));
```
[the remark this answers is not in the document]
Comment (GTC6244) [reply lit-api-server/src/accounts/decode_revert.rs]: Fixed in 20abdfc. Unknown revert data is now truncated to the first 64 bytes with the total length shown, e.g. `Unknown revert data (256 bytes): 0xabcd...`
